### DISCUSSION TITLE
ArcGIS 11.2 support

### DISCUSSION THREAD

**Discussion by ScottTexas** [Esri/arcgis-powershell-dsc#500] [General]
ArcGIS 11.2 was recently released.  Anyone know/tested if this will support an 10.9 to 11.2 upgrade?

**Comment by cameronkroeker** [comment]
Hello @ScottTexas Module v4.2.0 for 11.2 has been officially released and supports upgrading to 11.2:

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v4.2.0

You may want to consider a transitional upgrade to 10.9.1 prior:
https://enterprise.arcgis.com/en/get-started/latest/windows/considerations-for-upgrades-to-11-0-and-later.htm#ESRI_SECTION1_A452158A52624344BFF7CA63B58A60DA

https://www.esri.com/arcgis-blog/products/arcgis-enterprise/administration/consider-arcgis-enterprise-10-9-1-until-youre-ready-to-turn-it-up-to-11/

Thanks,
Cameron K.

**Reply by ScottTexas** [reply]
Awesome. Thank you!  yes we are already on 10.9.1

**Comment by ScottTexas** [comment]
I am new to DSC.  Anyone know if special characters in passwords still an issue with using DSC?   

I ask because when I try to use this latest 11.2 and previous 11.1 I get an error trying to use my assigned domain service account:  "Windows Event 10005"   "Product: Portal for ArcGIS 11.2 -- Error 28809. Could not validate the credentials for account ..."

I am sure the password is correct and I added single backslash before the special characters in password.

**Reply by cameronkroeker** [reply]
Hi @ScottTexas,

Which special characters are causing an issue (and are there consecutive special characters in a row)? Also, is this for the `ConfigData.Credentials.ServiceAccount.Password` attribute?

I have tested various special character, such as:

```
$#^&!%*@
```

Some special character combinations cause some characters to escape. Such as `^&&` gets converted to `^&`. To avoid this type of situation we pass in the `ConfigData.Credentials.ServiceAccount.Password` value with double quotes. Perhaps this doesn't work for all combinations. For example, during the silent install (On the target node where the installation is taking place to view what is being passed in, you can open task manager -> details tab -> enable command line column and find the setup.exe process):

```
<path to the ArcGIS Server setup download>\setup.exe /qb ACCEPTEULA=yes USER_NAME=mydomain\mydomainaccount PASSWORD="Pa$$w0rd#Pa$$w0rd^&&!"
```

One suggestion is to also try using encrypted password files (`ConfigData.Credentials.ServiceAccount.PasswordFilePath`).

Thanks,
Cameron K.

**Reply by ScottTexas** [reply]
I have "\\" and "=" as special characters ... trying to get it fixed but dealing with bureaucracy :(
yes ConfigData.Credentials.ServiceAccount.Password
I checked logs and the backslash looks like it was escaped ok but it looks like (maybe) the quotes are being passed as part of the password ... I'll keep trying

**Reply by cameronkroeker** [reply]
I think the `=` symbol could be problematic. The setup could be getting confused with it because there is an `=` symbol prior to the value. For example:

`setup.exe /qb PASSWORD="pass=word"`

I'm fairly certain this can be reproduced outside of DSC, by manually running a silent install.
